app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt

```diff
@@ -543,6 +544,91 @@ class DefaultTabsTrayControllerTest {
         }
     }
 
+    @Test
+    fun `GIVEN no tabs selected and the user is not in multi select mode WHEN the user long taps a tab THEN that tab will become selected`() {
+        trayStore = TabsTrayStore()
+        val selectionHolder: SelectionHolder<TabSessionState> = mockk {
+            every { selectedItems } returns emptySet()
+        }
+        val controller = spyk(createController())
+        val tab1 = TabSessionState(
+            id = "1",
+            content = ContentState(
+                url = "www.mozilla.com",
+            ),
+        )
+        val tab2 = TabSessionState(
+            id = "2",
+            content = ContentState(
+                url = "www.google.com",
+            ),
+        )
+        trayStore.dispatch(TabsTrayAction.ExitSelectMode)
+        trayStore.waitUntilIdle()
+
+        controller.handleMultiSelectClicked(tab1, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabSelected(tab1, "Tabs tray") }
+
+        controller.handleMultiSelectClicked(tab2, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabSelected(tab2, "Tabs tray") }
+    }
+
+    @Test
+    fun `GIVEN the user is in multi select mode and a tab is selected WHEN the user taps the selected tab THEN the tab will become unselected`() {
+        trayStore = TabsTrayStore()
+        val tab1 = TabSessionState(
+            id = "1",
+            content = ContentState(
+                url = "www.mozilla.com",
+            ),
+        )
+        val tab2 = TabSessionState(
+            id = "2",
+            content = ContentState(
+                url = "www.google.com",
+            ),
+        )
+        val selectionHolder: SelectionHolder<TabSessionState> = mockk {
+            every { selectedItems } returns setOf(tab1, tab2)
+        }
+        val controller = spyk(createController())
+        trayStore.dispatch(TabsTrayAction.EnterSelectMode)
+        trayStore.waitUntilIdle()
+
+        controller.handleMultiSelectClicked(tab1, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabUnselected(tab1) }
+
+        controller.handleMultiSelectClicked(tab2, selectionHolder, "Tabs tray")
+        verify(exactly = 1) { controller.handleTabUnselected(tab2) }
+    }
+
+    @Test
+    fun `GIVEN at least a tab is selected and the user is in multi select mode WHEN the user taps a tab THEN that tab will become selected`() {
+        trayStore = spyk(TabsTrayStore())
```

Comment:
@DreVla thanks for writing these tests, they're great!

One comment here that I think we should fix (after the monorepo migration is done): if we need to verify an action was sent to a store. We should use the [`CaptureActionsMiddleware`](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/support/test-libstate/src/main/java/mozilla/components/support/test/middleware/CaptureActionsMiddleware.kt) on the store to collect all the actions going the store, then assert with:

```kotlin
middleware.assertLastAction(TabsTrayAction.AddSelectTab::class) {
  assertEquals(tabs, it.tab)
}
```

Here is an [example test class](https://searchfox.org/mozilla-mobile/rev/d2356f2d8b7afc97efcf7db24eb4bc3e0ce246fd/firefox-android/android-components/components/feature/customtabs/src/test/java/mozilla/components/feature/customtabs/CustomTabIntentProcessorTest.kt) that might help as well.

Feel free to use the same bug number as well in the new repo to make the amendment.

cc: @MozillaNoah for reference too.